pandas/core/reshape/reshape.py

```diff
@@ -994,13 +995,15 @@ def _make_col_name(prefix, prefix_sep, level) -> str:
 
         dummy_cols = [_make_col_name(prefix, prefix_sep, level) for level in levels]
 
+    index: Optional[Index]
     if isinstance(data, Series):
         index = data.index
     else:
         index = None
 
     if sparse:
 
+        fill_value: Union[bool, float]
```

Comment:
could be int?

Reply:
yes, int is a structural subtype of float and hence the assignment on L1008 is valid for mypy.